File: Basics/0.5-For-Loops/Lab/clever_lily.py
lily_age = int(input())
washing_machine_price = float(input())
toy_price = int(input())
toys_count = 0
birthday_moneys = 0
birthday_moneys_sum = 0

for birthday in range(1, lily_age + 1):

    if birthday % 2 == 0:
        birthday_moneys += 10
        birthday_moneys_sum += birthday_moneys
        birthday_moneys_sum -= 1

    else:
        toys_count += 1

toys_moneys = toys_count * toy_price
birthday_moneys_sum += toys_moneys

diff = abs(birthday_moneys_sum - washing_machine_price)

if birthday_moneys_sum >= washing_machine_price:
    print(f"Yes! {diff:.2f}")
else:
    print(f"No! {diff:.2f}")

File: JavaScript/Basics/0.4-For-Loops/Lab/clever-lily.js
function cleverLily(input) {
    let years = Number(input[0]);
    let washingMachinePrice = Number(input[1]);
    let toyPrice = Number(input[2]);
    let toysCount = 0;
    let totalMoney = 0;
    let money = 0;
    for (i = 1; i <= years; i++) {
        if (i % 2 == 1) {
            toysCount += 1;
        } else {
            money += 10;
            totalMoney += money;
            totalMoney -= 1;
        }
    }
    totalMoney += toysCount * toyPrice;
    let diff = Math.abs(totalMoney - washingMachinePrice);
    if (totalMoney >= washingMachinePrice) {
        console.log(`Yes! ${diff.toFixed(2)}`)
    } else {
        console.log(`No! ${diff.toFixed(2)}`)
    }
}

cleverLily(["10", "170", "6"])
cleverLily(["21", "1570.98", "3"])